apps/web-client/src/lib/db/orders/reconciliation.ts

```diff
@@ -0,0 +1,88 @@
+import type { DB, ReconciliationOrder } from "./types";
+
+export async function getAllReconciliationOrders(db: DB): Promise<ReconciliationOrder[]> {
+	const result = await db.execO<ReconciliationOrder>(
+		"SELECT id, supplier_order_ids, customer_order_line_ids, created FROM reconciliation_order ORDER BY id ASC;"
+	);
+	return result;
+}
+
+export async function getReconciliationOrder(db: DB, id: number): Promise<ReconciliationOrder> {
+	const result = await db.execO<ReconciliationOrder>(
+		"SELECT id, supplier_order_ids, customer_order_line_ids, created FROM reconciliation_order WHERE id = ?;",
+		[id]
+	);
+	if (!result.length) {
+		throw new Error(`Reconciliation order with id ${id} not found`);
+	}
+	return result[0];
+}
+
+export async function createReconciliationOrder(db: DB, supplierOrderIds: string[]) {
+	if (!supplierOrderIds.length) {
+		throw new Error("Reconciliation order must be based on at least one supplier order");
+	}
+
+	const supplierOrderIdsJoined = supplierOrderIds.join(", ");
+
+	const tt = await db.execA(`INSERT INTO reconciliation_order (supplier_order_ids) VALUES (?) RETURNING id;`, [supplierOrderIdsJoined]);
```

Comment:
If we use this shape (have the `supplier_order_ids` be a comma separated string) we won't be able to join the supplier orders with the reconciliation data. I need to think some more about this. Will get back to it!

Reply:
After some pondering (and LLM chatting) I believe JSON support in sqlite is advanced enough that this layout/shape/architecture should not be a problem at all. We should change `supplier_order_ids` to be a full JSON array, and add a constraint for it to contain valid JSON, like `CHECK (json_valid(supplier_order_ids))`. Maybe also a check to make sure the JSON content is an array of at least length 1 (using `json_array_length`).

@coderabbitai can you confirm/critique my take on `supplier_order_ids`?

Reply:
Interesting. Given what I learned here I can improve the layout of the list of supplier orders we maintain for each customer order line: it could use the same strategy.